session/secrets/secretsprovider/file.go

```diff
@@ -0,0 +1,50 @@
+package secretsprovider
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/moby/buildkit/session/secrets"
+	"github.com/pkg/errors"
+)
+
+type FileSource struct {
+	ID       string
+	FilePath string
+}
+
+func NewFileStore(files []FileSource) (secrets.SecretStore, error) {
+	m := map[string]FileSource{}
+	for _, f := range files {
+		if f.ID == "" {
+			return nil, errors.Errorf("secret missing ID")
+		}
+		if f.FilePath == "" {
+			f.FilePath = f.ID
+		}
+		if _, err := os.Stat(f.FilePath); err != nil {
+			return nil, errors.Wrapf(err, "failed to stat %s", f.FilePath)
+		}
```

Comment:
maybe better to error out if the file is too large?

Reply:
Good point. Swarm secrets seem to be 500KB. Use same limit here? To support much bigger we would need to change the endpoint to stream.